samples-and-tests/just-test-cases/test/JobTest.java

```diff
@@ -0,0 +1,139 @@
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.junit.Test;
+
+import play.jobs.Every;
+import play.jobs.Job;
+import play.test.UnitTest;
+
+/**
+ * Unit tests for the {@link Job} class.
+ */
+public class JobTest extends UnitTest {
+
+    /**
+     * A Job class that is annotated with {@link Every} to run every second.
+     */
+    public static class TestJob extends Job {
+        private final AtomicLong totalRuns = new AtomicLong(0);
+        private volatile boolean throwException = false;
+
+        @Override
+        public void doJob() throws Exception {
+            totalRuns.getAndIncrement();
+
+            // To avoid logging an error every second, we only throw an
+            // exception when the relevant test case is running.
+            if (throwException) {
+                throwException = false;
+                throw new Exception("Throwing an exception");
+            }
+        }
+    }
+
+    /**
+     * A Job class that is annotated with {@link Every} to run every second.
+     */
+    @Every("1s")
+    public static class RunEverySecond extends Job {
+        private static final AtomicLong totalRuns = new AtomicLong(0);
+        private static volatile boolean throwException = false;
+
+        @Override
```

Comment:
I don't understand the recommendation.  This unit test is designed to test what happens when someone _doesn't_ override onException (the common case, and the case cited in [[#2060]](https://play.lighthouseapp.com/projects/57987/tickets/2060)).  If it overrides `onException`, then it would be testing something else.  Could you sketch out a bit more what you had in mind?